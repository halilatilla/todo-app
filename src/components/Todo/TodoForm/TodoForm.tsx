import { useState } from 'react'

import { addTodo, clearCompletedTodoList } from '@src/store/reducers/todoSlice'
import { useAppDispatch } from '@src/store/store'
import { Input, Button } from '@src/components'
import { isTextEmpty, removeWhiteSpace } from '@src/lib'

const TodoForm = () => {
  const [title, setTitle] = useState('')

  const dispatch = useAppDispatch()

  const onAddTodo = (event: any) => {
    event.preventDefault()
    if (isTextEmpty(title)) return
    dispatch(addTodo(removeWhiteSpace(title)))
    setTitle('')
  }

  const onClearCompleted = () => {
    dispatch(clearCompletedTodoList())
  }

  return (
    <form className="flex items-end justify-between gap-6" onSubmit={onAddTodo}>
      <div className="grid w-full grid-cols-4 items-end space-x-4">
        <Input
          value={title}
          placeholder="Add Todo"
          className="col-span-3"
          onChange={(e) => setTitle(e.currentTarget.value)}
        />
        <Button label="Add" className="px-5" type="submit" />
      </div>
      <Button label="clear completed" onClick={onClearCompleted} className="min-w-max px-4" />
    </form>
  )
}

export default TodoForm
