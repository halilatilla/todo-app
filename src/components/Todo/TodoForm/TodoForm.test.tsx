//import { render, screen } from '@testing-library/react'
import { render, screen, fireEvent } from '../../../../test-utils'

import TodoForm from './TodoForm'

describe('TodoForm', () => {
  it('renders correctly', () => {
    const { container } = render(<TodoForm />)
    expect(container).toMatchSnapshot()
  })

  it('should adds a new to-do', () => {
    render(<TodoForm />)

    const input = screen.getByPlaceholderText('Add Todo')
    const button = screen.getByText('Add')
    const todo = 'add a new to-do'

    fireEvent.change(input, { target: { value: todo } })
    button.click()

    expect(input).toHaveValue('')
  })
})
