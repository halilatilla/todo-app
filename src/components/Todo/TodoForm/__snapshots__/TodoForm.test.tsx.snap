// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoForm renders correctly 1`] = `
<div>
  <form
    class="flex items-end justify-between gap-6"
  >
    <div
      class="grid w-full grid-cols-4 items-end space-x-4"
    >
      <label
        class="block col-span-3"
      >
        <input
          class="input"
          placeholder="Add Todo"
          type="text"
          value=""
        />
      </label>
      <button
        class="button undefined px-5"
        type="submit"
      >
        Add
         
      </button>
    </div>
    <button
      class="button undefined min-w-max px-4"
      type="button"
    >
      clear completed
       
    </button>
  </form>
</div>
`;
